module-1/src/Components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,78 @@
+import { ReactNode } from "react";
+import { createPortal } from "react-dom";
+import "./Dialog.css";
+const mountElement: any = document.getElementById("overlays") || <div />;
+
+interface IDialogProps {
+  openDialog: boolean;
+  dialogTitle?: string;
+  children?: ReactNode;
+  closeCallback: () => void;
+  submitCallback: () => void;
+}
+
+export const Dialog = ({
+  openDialog = false,
+  dialogTitle,
+  children,
+  closeCallback,
+  submitCallback,
```

Comment:
This is very feature-specific, you should keep the props as generic as possible.